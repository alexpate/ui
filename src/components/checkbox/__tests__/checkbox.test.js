/* @flow */
/* eslint-env jest */
import {shallow} from 'enzyme';
import {testComponent} from 'utils/test/react';
import serializer from 'enzyme-to-json/serializer';

import Checkbox from '../checkbox';

expect.addSnapshotSerializer(serializer);

describe('components/checkbox', () => {
  describe('Checkbox', () => {
    const create = testComponent(Checkbox, () => ({
      size: 'medium',
      checked: true,
    }));

    it('should render shallow component ok', () => {
      const {element} = create();
      const result = shallow(element);
      expect(result).toMatchSnapshot();
    });

    it('should render a checkbox with label', () => {
      const {element} = create({
        label: 'A checkbox label',
      });

      const result = shallow(element);
      expect(result).toMatchSnapshot();
    });

    it('should render a checkbox partially checked', () => {
      const {element} = create({
        label: 'A checkbox label',
        checked: false,
        indeterminate: true,
      });
      const result = shallow(element);
      expect(result).toMatchSnapshot();
    });
  });
});
