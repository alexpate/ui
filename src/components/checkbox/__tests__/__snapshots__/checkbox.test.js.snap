// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render a checkbox partially checked 1`] = `
.glamor-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 14px;
  height: 14px;
}

.glamor-2 {
  display: inline-block;
  vertical-align: bottom;
  margin-left: 10px;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 14px;
  color: rgb(84,103,137);
  white-space: normal;
  text-overflow: initial;
  overflow: visible;
  white-space: pre-wrap;
  word-break: break-word;
}

.glamor-2:hover {
  text-decoration: inherit;
}

.glamor-2 strong {
  font-weight: 500;
}

<div
  aria-checked="mixed"
  className="root glamor-3"
  name={undefined}
  onClick={undefined}
  role="checkbox"
  style={Object {}}
>
  .glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 14px;
  height: 14px;
}

<div
    className="checkbox indeterminate glamor-0"
    style={Object {}}
  />
  .glamor-1 {
  display: inline-block;
  vertical-align: bottom;
  margin-left: 10px;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 14px;
  color: rgb(84,103,137);
  white-space: normal;
  text-overflow: initial;
  overflow: visible;
  white-space: pre-wrap;
  word-break: break-word;
}

.glamor-1:hover {
  text-decoration: inherit;
}

.glamor-1 strong {
  font-weight: 500;
}

<span
    className="glamor-0 glamor-1"
    color="navy600"
    onClick={undefined}
    size="small"
    style={Object {}}
  >
    A checkbox label
  </span>
</div>
`;

exports[`Checkbox should render a checkbox with label 1`] = `
.glamor-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 14px;
  height: 14px;
}

.glamor-2 {
  display: inline-block;
  vertical-align: bottom;
  margin-left: 10px;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 14px;
  color: rgb(84,103,137);
  white-space: normal;
  text-overflow: initial;
  overflow: visible;
  white-space: pre-wrap;
  word-break: break-word;
}

.glamor-2:hover {
  text-decoration: inherit;
}

.glamor-2 strong {
  font-weight: 500;
}

<div
  aria-checked={true}
  className="root glamor-3"
  name={undefined}
  onClick={undefined}
  role="checkbox"
  style={Object {}}
>
  .glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 14px;
  height: 14px;
}

<div
    className="checkbox checked glamor-0"
    style={Object {}}
  />
  .glamor-1 {
  display: inline-block;
  vertical-align: bottom;
  margin-left: 10px;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 14px;
  color: rgb(84,103,137);
  white-space: normal;
  text-overflow: initial;
  overflow: visible;
  white-space: pre-wrap;
  word-break: break-word;
}

.glamor-1:hover {
  text-decoration: inherit;
}

.glamor-1 strong {
  font-weight: 500;
}

<span
    className="glamor-0 glamor-1"
    color="navy600"
    onClick={undefined}
    size="small"
    style={Object {}}
  >
    A checkbox label
  </span>
</div>
`;

exports[`Checkbox should render shallow component ok 1`] = `
.glamor-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 14px;
  height: 14px;
}

<div
  aria-checked={true}
  className="root glamor-1"
  name={undefined}
  onClick={undefined}
  role="checkbox"
  style={Object {}}
>
  .glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 14px;
  height: 14px;
}

<div
    className="checkbox checked glamor-0"
    style={Object {}}
  />
</div>
`;
